pkg/endpoint/endpoint.go

```diff
@@ -2295,3 +2300,27 @@ func (e *Endpoint) PinDatapathMap() error {
 
 	return err
 }
+
+// SyncEndpointHeaderFile it bumps the current DNS History information for the
+// endpoint in the lxc_config.h file.
+func (e *Endpoint) SyncEndpointHeaderFile() error {
+	e.UnconditionalLock()
+	defer e.Unlock()
+
+	if e.dnsHistoryTrigger == nil {
+		t, err := trigger.NewTrigger(trigger.Parameters{
+			Name:              "sync_endpoint_header_file",
+			PrometheusMetrics: false,
+			MinInterval:       5 * time.Second,
+			TriggerFunc:       func(reasons []string) { e.writeHeaderfile(e.GetDirectoryPath()) },
+		})
+		if err != nil {
+			return fmt.Errorf(
+				"Sync Endpoint header file trigger for endpoint cannot be activated: %s",
+				err)
+		}
+		e.dnsHistoryTrigger = t
+	}
+	e.dnsHistoryTrigger.Trigger()
```

Comment:
You need to stop this trigger upon deletion of the endpoint with the `Shutdown` function.

Reply:
This is fixed ;-)